### ISSUE TITLE
Accessibility blocker: Commandbar: Index of commands no longer read out by screen reader

### ISSUE THREAD

**Issue by lukehumphrey** [OfficeDev/office-ui-fabric-react#8333]
This bug is preventing Project from passing an MS PWD usability test.
repro:

Edge > Narrator on.
Open Commandbar with commands in it and tab to select a command.
Expected: SR reads out command and index (x of y) commands.
Actual: (x of y) not read out. User doesn't know how many commands are in command bar.

Can this be fixed in the short-term?
Contact PM lukehum@microsoft.com for more qs.

**Comment by KevinTCoughlin** [comment]
Thanks for reporting this @lukehumphrey. It looks like the root element of `CommandBar` has `role="menubar"`. According to W3 ARIA specification, the role [menubar](https://www.w3.org/TR/wai-aria-1.1/#menubar) inherits from [menu](https://www.w3.org/TR/wai-aria-1.1/#menu) and requires owned elements be of role `menuitem`. 

The role `menuitem` supports state properties `aria-posinset` and `aria-setsize` which AT announces `X of Y` when applied.

Therefore this seems like a valid fix and one that we can provide. I will work on a patch now.

**Comment by KevinTCoughlin** [comment]
@lukehumphrey, you mention "Index of commands no longer read out by screen reader". Was this recently regressed to your knowledge? I will investigate git history.

**Comment by KevinTCoughlin** [comment]
@lukehumphrey it looks like this was removed in https://github.com/OfficeDev/office-ui-fabric-react/commit/cc7f252b7f921c8f2ba79551ea82fa6f93e36dda#diff-b7ad51933f12bb1d434bb4dd8cc5a969

@kenotron do you have context? I think `aria-posinset` and `aria-setsize` should be valid as long as the buttons have `role="menuitem"` as described in https://github.com/OfficeDev/office-ui-fabric-react/issues/8333#issuecomment-473114816.

**Comment by KevinTCoughlin** [comment]
@micahgodbolt can you advise as the author of https://github.com/OfficeDev/office-ui-fabric-react/issues/6232?

**Comment by kenotron** [comment]
If you looked at the linked bug... it was through some research about some bugs that cropped up with the component itself. The resolution isn't maybe ideal for this instance, but there was a workaround embedded in that statement:

> So with all of this research, and agreement from the the dev who initially added aria-setsize, I'm proposing we remove aria-setsize/posinset from command bar. (these values can always be added manually if required).
>
> Removing these properties will also remove a bug where we are mutating props to add these values in, so it'll be great to not have to do that anymore.

**Comment by micahgodbolt** [comment]
@KevinTCoughlin pos-in-set can be added to each command bar item if your scenario requires it. From what we've seen, adding pos-in-set caused more trouble for accessibility testing than it solved. 

Pos-in-set only needs to be set when not all items are present in the DOM at the same time (think pagination or virtualized list). Otherwise the screenreader has no trouble determining how many items are in the list by simply viewing the DOM.

**Comment by KevinTCoughlin** [comment]
Kudos to @micahgodbolt for supplying the original pen which I made a few modifications to here: https://codepen.io/kevintcoughlin/pen/ywxoEj?editors=1010

@lukehumphrey the Codepen above is a way to re-enable this announcement behavior for `CommandBar` by providing the props per item definition, i.e.

```tsx
      ...
      {
        'aria-posinset': 3,
        'aria-setsize': 7,
        key: 'share',
        name: 'Share',
        iconProps: {
          iconName: 'Share'
        },
        onClick: () => console.log('Share')
      },
      ...
```

This works well in my testing with Narrator + Edge:

```shell
1 of 7, 
 Upload, menu item,
Tooltip, https://dev.office.com/fabric.
2 of 7, 
 Share, menu item,
3 of 7, 
 Download, menu item,
4 of 7, 
More commands, button collapsed,
Sort, menu item,
5 of 7, 
Grid view, menu item,
6 of 7, 
Info, menu item,
7 of 7, 
Grid view, menu item,
Sort, menu item,
More commands, button collapsed,
Space
More commands, button expanded,
 Move to..., menu item,
1 of 3, 
 Copy to..., menu item,
2 of 3, 
 Rename..., menu item,
3 of 3,
```
